polyfill/intersection-observer.js

```diff
@@ -863,6 +863,30 @@ function getEmptyRect() {
 }
 
 
+/**
+ * Ensure that the result has all of the necessary fields of the DOMRect.
+ * Specifically this ensures that `x` and `y` fields are set.
+ *
+ * @param {?DOMRect} rect
+ * @return {?DOMRect}
+ */
+function fixDOMRect(rect) {
+  if (!rect || 'x' in rect) {
```

Comment:
Can you clarify what this check is for? Maybe add a comment?

Reply:
Done